hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHADBTransactionBufferImpl.java

```diff
@@ -56,13 +58,23 @@ private BatchOperation getCurrentBatchOperation() {
   @Override
   public <KEY, VALUE> void addToBuffer(
       Table<KEY, VALUE> table, KEY key, VALUE value) throws IOException {
-    table.putWithBatch(getCurrentBatchOperation(), key, value);
+    rwLock.readLock().lock();
```

Comment:
Are you sure we can use a read lock?  Unless `currentBatchOperation` is thread safe, we need a write lock here.

If we change this and `removeFromBuffer` to write lock, then everything uses the write lock.  We may change them to simply `synchronized`.

Reply:
@szetszwo I checked and found as below:
https://github.com/facebook/rocksdb/blob/b7e73501d82661455755b532dff8557e82cb1165/include/rocksdb/write_batch.h#L20-L23
From what I understand, the const method is the method that does not change a member variable of the class. This is exactly the case here we meet, the `flush` will change the `currentBatchOperation` while the `add/remove` will not, so I think the read lock is fine and it is good for performance.

Reply:
Good to know.  We have add some tests to verify it in a separated JIRA.